src/telemetry_poller.erl

```diff
@@ -0,0 +1,387 @@
+    }.
+
+parse_args(Args) ->
+    Measurements = proplists:get_value(measurements, Args, []),
+    validate_measurements(Measurements),
+    Period = proplists:get_value(period, Args, 5000),
+    validate_period(Period),
+    VMMeasurements = parse_vm_measurements(proplists:get_value(vm_measurements, Args, [])),
+
+    #{measurements => Measurements ++ VMMeasurements, period => Period}.
+
+-spec parse_vm_measurements(default | [vm_measurement()]) -> [measurement()].
+parse_vm_measurements([]) ->
+    [];
+parse_vm_measurements(default) ->
+    parse_vm_measurements(?DEFAULT_VM_MEASUREMENTS);
+parse_vm_measurements(Measurements) ->
+    __Measurements = sets:from_list(Measurements),
+    lists:map(fun parse_vm_measurement/1, sets:to_list(__Measurements)).
+
+-spec parse_vm_measurement(vm_measurement()) -> {telemetry_poller_vm, vm_measurement(), []} | no_return().
+parse_vm_measurement(Measurement) ->
+    SupportedMeasurement = lists:member(Measurement, ?DEFAULT_VM_MEASUREMENTS),
+    if SupportedMeasurement =:= false ->
+        erlang:error({badarg, "The specified measurement is not currently supported. Consider implementing a custom measurement."}, [Measurement]);
+        true ->
+            vm_measurement(Measurement)
+    end.
+
+-spec vm_measurement(vm_measurement()) -> measurement().
+vm_measurement(Function) ->
+    {telemetry_poller_vm, Function, []}.
+
+-spec schedule_measurement(non_neg_integer()) -> ok.
+schedule_measurement(CollectInMillis) ->
+    erlang:send_after(CollectInMillis, self(), collect), ok.
+
+-spec validate_period(term()) -> ok | no_return().
+validate_period(Period) when is_integer(Period), Period > 0 ->
+    ok;
+validate_period(Term) ->
+    erlang:error({badarg, "Expected period to be a positive integer"}, [Term]).
+
+-spec validate_measurements(term()) -> ok | no_return().
+validate_measurements(Measurements) when is_list(Measurements) ->
+    lists:foreach(fun validate_measurement/1, Measurements);
+validate_measurements(Term) ->
+    erlang:error({badarg, "Expected measurements to be a list"}, [Term]).
+
+-spec validate_measurement(term()) -> ok | no_return().
+validate_measurement({M, F, A}) when is_atom(M), is_atom(F), is_list(A) ->
+    ok;
+validate_measurement(Term) ->
+    erlang:error({badarg, "Expected measurement to be a valid MFA tuple"}, [Term]).
+
+-spec make_measurements_and_filter_misbehaving([measurement()]) -> [measurement()].
+make_measurements_and_filter_misbehaving(Measurements) ->
+    [Measurement || Measurement <- Measurements, make_measurement(Measurement) =/= error].
+
+-spec make_measurement(measurement()) -> measurement() | no_return().
+make_measurement(Measurement = {M, F, A}) ->
+    try erlang:apply(M, F, A) of
+        _ -> Measurement
+    catch
+        ?WITH_STACKTRACE(Class, Reason, Stacktrace)
+            ?LOG_ERROR("Error when calling MFA defined by measurement: ~p ~p ~p ."
+                        "Exception: class=~p reason=~p stacktrace=~p",
```

Comment:
So I have been working with telemetry and I found this format quite hard to read. Here is my proposal:

```erlang
"Error when calling MFA defined by measurement: ~p ~p ~p~n"
"Class=~p~nReason=~p~nStacktrace=~p~n"
```

What do you think?

I will also update the format on telemetry.

Reply:
Agreed. That's much nicer to read.